var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -103,3 +105,28 @@ def post_install(self):
         install('control/magma_threadsetting.h', self.prefix.include)
         install('control/pthread_barrier.h', self.prefix.include)
         install('control/magma_internal.h', self.prefix.include)
+
+    test_src_dir = 'example'
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([self.test_src_dir])
+
+    def test(self):
+        test_dir = join_path(self.install_test_root, self.test_src_dir)
+        with working_dir(test_dir, create=False):
+            magma_dir = 'MAGMADIR={0}'.format(self.prefix)
+            cuda_dir = 'CUDADIR={0}'.format(self.spec['cuda'].prefix)
+            blas_dir = 'OPENBLASDIR={0}'.format(self.spec['blas'].prefix)
+            test_msg = 'MAGMA smoke test'
+            make(magma_dir, cuda_dir, blas_dir, 'c')
+            self.run_test('./example_sparse', purpose=test_msg)
+            self.run_test('./example_sparse_operator', purpose=test_msg)
+            self.run_test('./example_v1', purpose=test_msg)
+            self.run_test('./example_v2', purpose=test_msg)
+            if '+fortran' in self.spec:
+                make(magma_dir, cuda_dir, blas_dir, 'fortran')
+                self.run_test('./example_f', purpose=test_msg)
```

Comment:
Can you provide a different test message for each invocation of `run_test`?

The reason for providing the `purpose` argument is to make `gripping the test output -- to find specific test parts -- easier.  If they are all named the same thing it could get confusing.

Otherwise, you can leave the `purpose` off and we will at least see the command line equivalent for each test part.

Reply:
Thanks for pointing it out! I failed to see the importance of the purpose message actually, but now I understand why it makes sense to have a different message